src/Medidata.ZipkinTracer.Core/ZipkinClient.cs

```diff
@@ -19,109 +20,155 @@ public class ZipkinClient : ITracerClient
         public ZipkinClient(ITraceProvider traceProvider, ILog logger, IZipkinConfig zipkinConfig, ISpanCollectorBuilder spanCollectorBuilder)
         {
             this.logger = logger;
-            isTraceOn = true;
 
-            if ( logger == null || !IsConfigValuesValid(zipkinConfig) || !IsTraceProviderValidAndSamplingOn(traceProvider))
+            isTraceOn = logger != null && IsConfigValuesValid(zipkinConfig) && IsTraceProviderValidAndSamplingOn(traceProvider);
+
+            if (!isTraceOn)
+                return;
+
+            try
             {
-                isTraceOn = false;
+                spanCollector = spanCollectorBuilder.Build(
+                    new Uri(zipkinConfig.ZipkinBaseUri),
+                    int.Parse(zipkinConfig.SpanProcessorBatchSize),
+                    logger);
+
+                spanTracer = new SpanTracer(
+                    spanCollector,
+                    new ServiceEndpoint(),
+                    zipkinConfig.GetNotToBeDisplayedDomainList(),
+                    zipkinConfig.Domain,
+                    zipkinConfig.ServiceName);
+
+                this.traceProvider = traceProvider;
             }
-
-            if (isTraceOn)
+            catch (Exception ex)
             {
-                try
-                {
-                    spanCollector = spanCollectorBuilder.Build(
-                        new Uri(zipkinConfig.ZipkinBaseUri),
-                        int.Parse(zipkinConfig.SpanProcessorBatchSize),
-                        logger);
-
-                    spanTracer = new SpanTracer(
-                        spanCollector,
-                        new ServiceEndpoint(),
-                        zipkinConfig.GetNotToBeDisplayedDomainList(),
-                        zipkinConfig.Domain,
-                        zipkinConfig.ServiceName);
-
-                    this.traceProvider = traceProvider;
-                }
-                catch (Exception ex)
-                {
-                    logger.Error("Error Building Zipkin Client Provider", ex);
-                    isTraceOn = false;
-                }
+                logger.Error("Error Building Zipkin Client Provider", ex);
+                isTraceOn = false;
```

Comment:
@jfeltesse-mdsol @lschreck-mdsol Yes, it will control subsequent calls on different methods. If the the spantracer and spancollector failed to initialized correctly then this flag will prevent any zipkin methods to execute.